src/page-modules/assistant/utils.ts

```diff
@@ -1,38 +1,50 @@
+import {
+  DepartureDate,
+  DepartureDateState,
+} from '@atb/components/departure-date-selector';
+import { TransportModeFilterState } from '@atb/components/transport-mode-filter/types';
+import { filterToQueryString } from '@atb/components/transport-mode-filter/utils';
 import { GeocoderFeature } from '@atb/page-modules/departures';
-
-export type FromToQuery = {
-  fromId: string;
-  fromName: string;
-  fromLon: string;
-  fromLat: string;
-  fromLayer: string;
-  toId: string;
-  toName: string;
-  toLon: string;
-  toLat: string;
-  toLayer: string;
-};
+import { TripQuery, TripQuerySchema } from '@atb/page-modules/assistant';
 
 export const featuresToFromToQuery = (
-  from: GeocoderFeature | undefined,
-  to: GeocoderFeature | undefined,
-): FromToQuery | null => {
-  if (!from || !to) return null;
+  from: GeocoderFeature,
+  to: GeocoderFeature,
+) => {
   return {
     fromId: from.id,
-    fromName: from.name,
     fromLon: from.geometry.coordinates[0].toString(),
     fromLat: from.geometry.coordinates[1].toString(),
     fromLayer: from.layer,
     toId: to.id,
-    toName: to.name,
     toLon: to.geometry.coordinates[0].toString(),
     toLat: to.geometry.coordinates[1].toString(),
     toLayer: to.layer,
   };
 };
 
-export const parseLayerQueryString = (layer: string): 'address' | 'venue' => {
-  if (layer === 'venue') return 'venue';
-  return 'address';
+export const createTripQuery = (
+  fromFeature: GeocoderFeature,
+  toFeature: GeocoderFeature,
+  transportModeFilter?: TransportModeFilterState,
+  departureDate?: DepartureDate,
+): TripQuery => {
+  const filter =
+    transportModeFilter && filterToQueryString(transportModeFilter);
+  const arriveBy = departureDate?.type === DepartureDateState.Arrival;
+  const departBy = departureDate?.type === DepartureDateState.Departure;
+  const fromToQuery = featuresToFromToQuery(fromFeature, toFeature);
+
+  return {
+    ...(filter ? { filter } : {}),
```

Comment:
Isn't this the same? 🤔
```suggestion
    filter,
```

Reply:
The filter could be null, but I've updated it to be either a string or undefined which makes more sense in this case 👍